src/tunnel.cc

```diff
@@ -416,6 +416,9 @@ TunnelStateData::TunnelStateData(ClientHttpRequest *clientRequest) :
     AsyncCall::Pointer timeoutCall = commCbCall(5, 4, "tunnelTimeout",
                                      CommTimeoutCbPtrFun(tunnelTimeout, this));
     commSetConnTimeout(client.conn, Config.Timeout.lifetime, timeoutCall);
+
+    // could have been started already
+    request->hier.startPeerClock();
```

Comment:
Please use a trailing comment. It reads better IMHO because the preceding code creates context, explaining _what_ could have been started.

```suggestion
    request->hier.startPeerClock(); // could have been started already
```